### PR TITLE
Add field name sanitization

### PR DESCRIPTION
**Pull request by CodingParsley**
What
---------
BigQuery dataset only allows underscore as special character for field names, and field names have to start with a letter or underscore.

cc: @C0urante

### CODE REVIEW

**Comment by xvrl** [comment]
@criccomini just FYI, @CodingParsley is working for Confluent, I'm assuming we must have some kind of corporate CLA already in place given that we've submitted patches in the past.

**Comment by apoorvmittal10** [comment]
@criccomini Please if you can help closing this PR.

File: kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/utils/FieldNameSanitizer.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class FieldNameSanitizer {
+
+  // replace all non-letter, non-digit characters with underscore
+  // append underscore in front of name if it does not begin with letter or underscore
+  private static String sanitizeName(String name) {
+    name = name.replaceAll("[^a-zA-Z0-9_]", "_");
+    if (name.matches("^[^a-zA-Z_]+.*")) {
```
Comment (mtagle) [inline kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/utils/FieldNameSanitizer.java]: nit, but is there a need for the + here?

File: kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/utils/FieldNameSanitizerTest.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+          }});
+          put("A-2", new HashMap<String, Object>() {{
+            put("=/B.3", "hello B3");
+            put("B./4", "hello B4");
+            put("2A/", "hello B5");
+            put("3A/", "hello B6");
+          }});
+        }};
+  }
+
```
Comment (mtagle) [inline kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/utils/FieldNameSanitizerTest.java]: nit: remove extra empty line here.

File: kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/utils/FieldNameSanitizerTest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    assertTrue(((HashMap)(testMap.get("A_2"))).containsKey("__B_3"));
+    assertTrue(((HashMap)(testMap.get("A_2"))).containsKey("B__4"));
+
+    assertEquals(((HashMap)(testMap.get("A_2"))).get("__B_3"), "hello B3");
+    assertEquals(((HashMap)(testMap.get("A_2"))).get("B__4"), "hello B4");
+
+    assertEquals(((HashMap)(testMap.get("A_2"))).get("_2A_"), "hello B5");
+    assertEquals(((HashMap)(testMap.get("A_2"))).get("_3A_"), "hello B6");
+
```
Comment (mtagle) [inline kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/utils/FieldNameSanitizerTest.java]: remove line

**Comment by CodingParsley** [comment]
fixed nit

**Comment by CodingParsley** [comment]
@apoorvmittal10 @xvrl I saw the option of signing SLA to contribute on behalf an employer so I signed it.

**Comment by criccomini** [comment]
@CodingParsley I think this is the issue:

> Naya Chen seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please add the email address used for this commit to your account.

Your commits aren't using the same email as your GH user.

**Comment by mtagle** [comment]
To clarify, it looks like your commits are using an email address that isn't associated to your github account. If you add that email address to your github account that should fix everything up.

**Comment by apoorvmittal10** [comment]
@CodingParsley Please close this PR.

**Pull request closed without merging**